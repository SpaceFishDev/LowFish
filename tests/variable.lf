i32 main()
{
	i32 x = 55;
	i8 y = 5;
	i32 z = x;
}
