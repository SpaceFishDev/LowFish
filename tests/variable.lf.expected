<NODE( 'PROGRAM', PROGRAM )>
{
    <NODE( 'FUNCTION', FUNCTION )>
    {
        <NODE( 'i32', TYPE )>
        <NODE( 'main', TOKEN_NODE )>
        <NODE( 'BLOCK', BLOCK )>
        {
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i32', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'x', TOKEN_NODE )>
                        <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                        {
                            <NODE( '55', TOKEN_NODE )>
                        }
                    }
                }
            }
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i8', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'y', TOKEN_NODE )>
                        <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                        {
                            <NODE( '5', TOKEN_NODE )>
                        }
                    }
                }
            }
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i32', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'z', TOKEN_NODE )>
                        <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                        {
                            <NODE( 'x', TOKEN_NODE )>
                        }
                    }
                }
            }
        }
    }
}
